### PR TITLE
Add separate cluster management commits to onit

### PR DESCRIPTION
**Pull request by kuujo**
#383 

This PR adds separate commands for managing test clusters via `onit`. The command now stores configurations for clusters, supports multiple clusters, allows simulators to be added to/removed from clusters, etc.

To create a cluster:
```
onit create cluster
```

Once the cluster has been created, you will see it in the clusters list:
```
onit get clusters
```

It will also be set as the default cluster:
```
onit get cluster
```

To add a simulator to the cluster:
```
onit add simulator
```

When the simulator is added, the onos-config stores will be updated and the pods will be redeployed with the updated configuration.

Similarly, simulators can be removed:
```
onit remove simulator
```

Once the cluster has been setup, tests can be run as normal:
```
onit run test-integration-test
```

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -20,13 +20,11 @@ For Minikube, ensure you build the image in the Minikube Docker context:
 ```
 
 For kind, you must manually `load` the `onos-config` image and the integration tests image 
-into the Kubernetes cluster each time the tests are modified:
+into the Kubernetes cluster each time the tests are modified. The `make kind` target will
```
Comment (Andrea-Campanella) [inline docs/testing.md]: you might want to add that if you make changes to the simulators or cli or debug images you need to load those manually and provide the `kind load docker-image onosproject/<image>:<tag>` example

File: test/runner/config.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+	configs := []string{}
+	filepath.Walk(deviceConfigsPath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+
+		ext := filepath.Ext(info.Name())
+		if ext == ".json" {
+			configs = append(configs, info.Name()[:len(info.Name())-len(ext)])
+		}
```
Comment (Andrea-Campanella) [inline test/runner/config.go]: shoudl we drop an error if we don't have `.json` here ?

File: docs/testing.md (hunk 1 above)
Comment (kuujo) [reply docs/testing.md]: @Andrea-Campanella the `make kind` target runs `kind load ...`

File: test/runner/config.go (hunk 2 above)
Comment (kuujo) [reply test/runner/config.go]: I think we should allow for something like `.md` files to be in the config directories

**Comment by tomikazi** [comment]
This is awesome Jordan!

File: docs/testing.md (hunk 1 above)
Comment (Andrea-Campanella) [reply docs/testing.md]: @kuujo +1

**Comment by kuujo** [comment]
Added additional commands for viewing the state of clusters and tests. Most commands are now documented in the updated `testing.md` documentation.

**Pull request merged**